tests/testapp/tests/test_syncsession_methods.py

```diff
@@ -0,0 +1,251 @@
+import factory
+
+from django.db import connection
+from django.test import TestCase
+from facility_profile.models import Facility
+from morango.controller import MorangoProfileController
+from morango.syncsession import SyncClient
+from morango.models import Buffer, DatabaseIDModel, InstanceIDModel, Store, RecordMaxCounter, RecordMaxCounterBuffer
+from .helpers import create_dummy_store_data, create_buffer_and_store_dummy_data
+
+class FacilityModelFactory(factory.DjangoModelFactory):
+
+    class Meta:
+        model = Facility
+
+    name = factory.Sequence(lambda n: "Fac %d" % n)
+
+
+class QueueStoreIntoBufferTestCase(TestCase):
+
+    def setUp(self):
+        self.data = create_dummy_store_data()
+        self.filter_prefixes = []
+        self.fsics = {}
+
+    def test_all_fsics(self):
+        self.fsics = {self.data['group1_id'].id: 0, self.data['group2_id'].id: 0}
+        self.data['sc']._queue_into_buffer(self.filter_prefixes, self.fsics)
+        buffer_ids = Buffer.objects.values_list('model_uuid', flat=True)
+        rmcb_ids = RecordMaxCounterBuffer.objects.values_list('model_uuid', flat=True)
+        for i in self.data['group1_c1']:
+            self.assertIn(i.id, buffer_ids)
+            self.assertIn(i.id, rmcb_ids)
+        for i in self.data['group1_c2']:
+            self.assertIn(i.id, buffer_ids)
+            self.assertIn(i.id, rmcb_ids)
+        for i in self.data['group2_c1']:
+            self.assertIn(i.id, buffer_ids)
+            self.assertIn(i.id, rmcb_ids)
```

Comment:
My reason for doing several loops was because in case one of the data groups is not actually in the list of ids, then I actually know which group is failing. Since each group has different partition values and last_saved_bys. Instead of grouping them together and having it all fail if only one of the groups is wrong.